stream/src/main/scala/org/apache/pekko/stream/scaladsl/Flow.scala

```diff
@@ -2267,8 +2277,17 @@ trait FlowOps[+Out, +Mat] {
    *
    * @see [[#splitWhen]]
    */
-  def splitWhen(p: Out => Boolean): SubFlow[Out, Mat, Repr, Closed] =
-    splitWhen(SubstreamCancelStrategy.drain)(p)
+  def splitWhen(p: Out => Boolean): SubFlow[Out, Mat, Repr, Closed] = {
```

Comment:
This would be against idiomatic design of Akka Streams in the sense that `SupervisionDecider`  is never supplied as a parameter, rather its always supplied by the `Attributes` mechanism